### PR TITLE
Add release note and update versions to v2.3

### PR DESCRIPTION
**Pull request by kvartet**
remember to update the release date :)

Documentation update needed:

+ Retiarii doc
+ 3rd-party training service doc
+ setting prefix URL doc
+ Tensorflow slim pruner doc

### CODE REVIEW

File: docs/en_US/Release.rst [hunk 1]
```diff
@@ -5,6 +5,93 @@
...
+* Support LSQ quantizer (#3503, thanks the external contributor @chenbohua3)
+* Improve APIs for iterative pruners (#3507 #3688)
+
+Training service & Rest
+"""""""""""""""""""""""
+
+* Support 3rd-party training service (#3662 #3726)
+* Support setting prefix URL (#3625 #3674 #3672 #3643)
+* Improve NNI manager logging (#3624)
+* Remove tensorboard code on nnictl (#3613)
```
Comment (scarlett2018) [inline docs/en_US/Release.rst:42]: add an explanation about why we remove it?

File: docs/en_US/Release.rst [hunk 2]
```diff
@@ -5,6 +5,93 @@
...
+* Improve search parameters on trial detail page (#3651 #3723 #3715)
+* Make selected trials consistent after auto-refresh in detail table (#3597)
+* Add trial stdout button on local mode (#3653 #3690)
+
+Examples & Documentation
+""""""""""""""""""""""""
+
+* Convert all trial examples' from config v1 to config v2 (#3721 #3733 #3711 #3600)
+* Add new jupyter notebook examples (#3599 #3700)
+* Add pipeline badge in README (#3589)
```
Comment (scarlett2018) [inline docs/en_US/Release.rst:62]: this is not document or example, but developer/code health management. can move it to others.

File: docs/en_US/Release.rst [hunk 3]
```diff
@@ -5,6 +5,93 @@
...
+
+Others
+""""""
```
Comment (scarlett2018) [inline docs/en_US/Release.rst:65]: Others --> DEV excellent

**Review by scarlett2018 — CHANGES_REQUESTED** [review]

**Comment by scarlett2018** [comment]
I had also made a quick fix on the retiarii part and merged to the PR directly.

**Review by scarlett2018 — COMMENTED** [review]
update link

**Comment by kvartet** [comment]
@scarlett2018 updated all.

File: docs/en_US/Release.rst [hunk 4]
```diff
@@ -5,6 +5,94 @@
...
+* Make selected trials consistent after auto-refresh in detail table (#3597)
+* Add trial stdout button on local mode (#3653 #3690)
+
+Examples & Documentation
+""""""""""""""""""""""""
+
+* Convert all trial examples' from config v1 to config v2 (#3721 #3733 #3711 #3600)
+* Add new jupyter notebook examples (#3599 #3700)
+
+Dev Excellent
```
Comment (ultmaster) [inline docs/en_US/Release.rst:64]: What is Dev Excellent

**Review by ultmaster — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -26,9 +26,9 @@ The tool manages automated machine learning (AutoML) experiments, **dispatches a
 * ML Platform owners who want to **support AutoML in their platform**.
 
 ## **What's NEW!** &nbsp;<a href="#nni-released-reminder"><img width="48" src="docs/img/release_icon.png"></a>
-* **New release**: [v2.2 is available](https://github.com/microsoft/nni/releases) - _released on April-26-2021_
-* **New demo available**: [Youtube entry](https://www.youtube.com/channel/UCKcafm6861B2mnYhPbZHavw) | [Bilibili 入口](https://space.bilibili.com/1649051673) - _last updated on May-26-2021_
 
+* **New release**: [v2.3 is available](https://github.com/microsoft/nni/releases) - _released on April-26-2021_
```
Comment (scarlett2018) [inline README.md:30]: >_released on April-26-2021

6/15/2021

**Review by J-shang — APPROVED** [review]

**Review by scarlett2018 — APPROVED** [review]

**Review by QuanluZhang — APPROVED** [review]

**Review by SparkSnail — APPROVED** [review]

**Review by liuzhe-lz — APPROVED** [review]

**Pull request merged**